pkg/services/hcloud/server/server_test.go

```diff
@@ -339,3 +341,111 @@ var _ = Describe("Test ValidateLabels", func() {
 		}),
 	)
 })
+
+var _ = Describe("Test handleRateLimit", func() {
+	type testCaseHandleRateLimit struct {
+		hm           *infrav1.HCloudMachine
+		err          error
+		functionName string
+		errMsg       string
+		expectError  error
+		conditionSet bool
+	}
+
+	DescribeTable("Test handleRateLimit",
+		func(tc testCaseHandleRateLimit) {
+			err := handleRateLimit(tc.hm, tc.err, tc.functionName, tc.errMsg)
+			if tc.expectError != nil && hcloud.IsError(err, hcloud.ErrorCodeRateLimitExceeded) {
+				Expect(err).To(MatchError(tc.expectError))
+				Expect(isPresentAndFalseWithReason(tc.hm, infrav1.HetznerAPIReachableCondition, infrav1.RateLimitExceededReason)).To(BeTrue())
+			}
+			if tc.expectError != nil {
+				Expect(err).To(MatchError(tc.expectError))
+			}
+			if tc.expectError == nil {
+				Expect(err).To(BeNil())
+			}
+			if tc.conditionSet {
+				Expect(isPresentAndFalseWithReason(tc.hm, infrav1.HetznerAPIReachableCondition, infrav1.RateLimitExceededReason)).To(BeTrue())
+			} else {
+				Expect(isPresentAndFalseWithReason(tc.hm, infrav1.HetznerAPIReachableCondition, infrav1.RateLimitExceededReason)).To(BeFalse())
```

Comment:
you expect this condition to not be set at all I think. Right now you check only that it is not set on false with a specific reason